### PR TITLE
435 handle b64 in xml

### PR DESCRIPTION
**Pull request by Orta21**
Ref: 435

### Description

Revert back to stream download into s3

### Release Plan

- [ ] ASAP

### CODE REVIEW

File: packages/lambdas/src/document-downloader.ts [hunk 1]
```diff
@@ -77,15 +77,19 @@ export const handler = Sentry.AWSLambda.wrapHandler(
...
 
-    pass.on("data", chunk => {
+    writeStream.on("data", chunk => {
       downloadedDocument += chunk;
     });
 
-    pass.on("end", () => {
+    writeStream.on("end", () => {
       console.log("Finished downloading document");
     });
```
Comment (leite08) [inline packages/lambdas/src/document-downloader.ts:94]: I think we don't need that

Comment (Orta21) [reply packages/lambdas/src/document-downloader.ts:94]: I thought we'd still want it instead of redownloading to the doc from s3 but i can remove. lmk

**Review by leite08 — COMMENTED** [review]
I know it'll add time to the fix, but I suggest we build some integration test w/ Jest so we can run code local that uses functions on `document-downloader.ts` to validate the download is working, even if its a test/script to run manually w/ a CW binary filename

File: packages/lambdas/src/document-downloader.ts (hunk 1 above)
Comment (leite08) [reply packages/lambdas/src/document-downloader.ts:94]: oh, gotcha, this is not being used to dowload, just to process the XML, right?

Comment (leite08) [reply packages/lambdas/src/document-downloader.ts:94]: yeah, got it... it makes sense to keep this 👍

File: packages/lambdas/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+const config: JestConfigWithTsJest = {
+  preset: "ts-jest",
+  testEnvironment: "node",
+  verbose: true,
+  testPathIgnorePatterns: ["/node_modules/", "/dist/"],
+  testMatch: isE2E
+    ? ["**/__tests__/**/(*.)+(spec|test).e2e.[jt]s?(x)"]
+    : ["**/__tests__/**/(*.)+(spec|test).[jt]s?(x)"],
+  setupFilesAfterEnv: ["./src/__tests__/env-setup.ts"],
```
Comment (leite08) [inline packages/lambdas/jest.config.ts:20]: This is only needed if tests require env variables to be present to work, which I don't think its the case on this package (we don't even have the `.env.test` being included on `env-setup.ts`).

File: packages/lambdas/src/__tests__/document-downloader.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    });
+
+    const uploadResult = await promise;
+
+    expect(uploadResult).toBeTruthy();
+    expect(uploadResult.Key).toEqual(file.fileName);
+
+    const { size, contentType } = await getFileInfoFromS3(uploadResult.Key, uploadResult.Bucket);
+
+    expect(size).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (Orta21) [reply packages/lambdas/src/__tests__/document-downloader.test.ts]: I added it as a field on the docRef to reference

File: packages/lambdas/src/__tests__/document-downloader.test.ignore.ts [hunk 4]
```diff
@@ -10,10 +10,11 @@ const cwOrgCertificate = ``;
 const cwOrgPrivateKey = ``;
 const bucketName = ``;
 
-const file = {
+const docRef = {
```
Comment (leite08) [inline packages/lambdas/src/__tests__/document-downloader.test.ignore.ts:13]: NABD, but maybe we can add a JSDoc here indicating that these should be updated before running the tests

**Review by leite08 — APPROVED** [review]
LGTM, just a small nit on adding a comment/todo on the integration test

File: packages/lambdas/package.json [hunk 5]
```diff
@@ -28,6 +28,7 @@
     "@sentry/serverless": "^7.46.0",
     "aws-sdk": "^2.1248.0",
     "axios": "^1.4.0",
+    "dotenv": "^16.3.1",
```
Comment (leite08) [inline packages/lambdas/package.json:31]: NABD, but I think this can be moved to `devDependencies`, no?

**Review by leite08 — APPROVED** [review]

**Pull request merged**